src/overlays/actors/ovl_En_Wonder_Talk/z_en_wonder_talk.c

```diff
@@ -26,15 +29,228 @@ const ActorInit En_Wonder_Talk_InitVars = {
+                    this->actionFunc = func_80B391CC;
+                    break;
+                case 2:
+                    switch (globalCtx->msgCtx.choiceIndex) {
+                        case 0:
+                            if (LINK_IS_CHILD) {
+                                // I'm still a child!
+                                osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ まだコドモなの！ ☆☆☆☆☆ \n" VT_RST);
+                                this->actor.textId = 0x5001;
+                            } else {
+                                // I'm an adult. .. ..
+                                osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ アダルトなの。。。 ☆☆☆☆☆ \n" VT_RST);
+                                this->actor.textId = 0x5003;
+                            }
+                            break;
+                        case 1:
+                            // Out!
+                            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ はずれ！ ☆☆☆☆☆ \n" VT_RST);
+                            this->actor.textId = 0x5004U;
+                            break;
+                    }
+
+                    this->unk_156 = 6;
+                    func_8010B720(globalCtx, this->actor.textId);
+                    this->actionFunc = func_80B391CC;
+                    break;
+                case 3:
+                    func_80106CCC(globalCtx);
+                    if (this->unk_164 == 0) {
+                        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_POH, this->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 2);
+                        this->unk_164 = 1;
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B391CC.s")
+                    this->actionFunc = func_80B391CC;
+                    break;
+                case 5:
+                    func_80106CCC(globalCtx);
+                    if (this->unk_164 == 0) {
+                        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_POH, this->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 3);
+                        this->unk_164 = 1;
+                    }
+                    this->actionFunc = func_80B391CC;
+                    break;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B3943C.s")
+void EnWonderTalk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnWonderTalk* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B395F0.s")
+    if (this->unk_158 != 0) {
+        this->unk_158 -= 1;
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->actor, this->fHeight);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/EnWonderTalk_Update.s")
+    if (BREG(0) != 0) {
+        if (this->unk_15A) {
```

Comment:
unk_15A isn't a boolean so `!= 0` should be used here.
```suggestion
        if (this->unk_15A != 0) {
```